### PR TITLE
Updated initialized this object in structs

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Fixed issue with `RayStep` and `Vector3Smoothed` struct not initializing this() before assigning values in constructor.

Also public fields were not PascalCase.

### CODE REVIEW

File: Assets/HoloToolkit/Input/Scripts/Focus/RayStep.cs [hunk 1]
```diff
@@ -9,43 +9,43 @@ namespace HoloToolkit.Unity
     [Serializable]
     public struct RayStep
     {
-        public RayStep(Vector3 origin, Vector3 terminus)
+        public RayStep(Vector3 origin, Vector3 terminus) : this()
```
Comment (keveleigh) [inline Assets/HoloToolkit/Input/Scripts/Focus/RayStep.cs]: This looks like a good change, but, to help me understand, what is this fixing in this case? I was under the impression ` : this()` was only needed if not all properties are set inside the constructor. In this case, all four are set.

Comment (StephenHodgson) [reply Assets/HoloToolkit/Input/Scripts/Focus/RayStep.cs]: `this` is required _because_ we're setting the values in the constructor.

Idk, I just keep getting a compiler warning about it, and figured we should heed the warnings.

Comment (keveleigh) [reply Assets/HoloToolkit/Input/Scripts/Focus/RayStep.cs]: Do you happen to have the warning? I can't seem to repro it.

Definitely agreed on fixing warnings though :)

Comment (StephenHodgson) [reply Assets/HoloToolkit/Input/Scripts/Focus/RayStep.cs]: image

Comment (Railboy) [reply Assets/HoloToolkit/Input/Scripts/Focus/RayStep.cs]: Better safe than sorry. Though I can't seem to raise the warning either - I've tried 4.6, 3.5, 2.0 and 2.0 subset.

Comment (NeerajW) [reply Assets/HoloToolkit/Input/Scripts/Focus/RayStep.cs]: Wow could not repro either :) Perhaps Stephen has some flags turned on in VS?

**Comment by StephenHodgson** [comment]
Here's another struct that needs the same treatment
image

**Comment by NeerajW** [comment]
@keveleigh can we get this merged?

**Pull request merged**